client/post-editor/editor-ground-control/index.jsx

```diff
@@ -55,20 +60,71 @@ export default React.createClass( {
 			onSaveDraft: noop,
 			post: null,
 			savedPost: null,
-			site: {}
+			site: {},
+			user: null,
+			userUtils: null,
 		};
 	},
 
+	componentDidMount: function() {
+		if ( ! config( 'email_verification_gate' ) ) return;
+
+		this.props.user
+			&& this.props.user
+			.on( 'change', this.updateNeedsVerification )
+			.on( 'verify', this.updateNeedsVerification );
+	},
+
+	componentWillUnmount: function() {
+		if ( ! config( 'email_verification_gate' ) ) return;
+
+		this.props.user
+			&& this.props.user
+			.off( 'change', this.updateNeedsVerification )
+			.off( 'verify', this.updateNeedsVerification );
+	},
+
+	updateNeedsVerification: function() {
+		this.setState( {
+			needsVerification: config( 'email_verification_gate' )
```

Comment:
When would we ever reach here that `config( 'email_verification_gate' )` is not true?